inspec/a2-api-integration/controls/iam_v2.rb

```diff
@@ -803,28 +771,50 @@
     end
 
     context "when there are no rules but the project exists" do
-      describe "GET /iam/v2beta/projects/:id/rules" do
-        it "returns an empty list" do
-          resp = automate_api_request("/apis/iam/v2beta/projects/#{CUSTOM_PROJECT_ID}/rules")
-          expect(resp.http_status).to eq 200
-          expect(resp.parsed_response_body[:rules]).to eq([])
-        end
+      before(:all) do
+        resp = automate_api_request("/apis/iam/v2beta/projects",
+         http_method: 'POST',
+         request_body: {
+           id: CUSTOM_PROJECT_ID,
+           name: "display name !#$#"
+         }.to_json
+       )
+       expect(resp.http_status).to eq 200
+     end
+
+     after(:all) do
+       resp = automate_api_request("/apis/iam/v2beta/projects/#{CUSTOM_PROJECT_ID}", http_method: 'DELETE')
+       expect(resp.http_status.to_s).to match(/200|404/)
+     end
+
+      it "GET /iam/v2beta/projects/:id/rules returns an empty list" do
```

Comment:
when there are no rules?

Reply:
Yes, that's the context block it's in in line 773:

```
context "when there are no rules but the project exists" do
```

Setting up preconditions via context is a core part of Rspec / Inspec. Otherwise, we'd need to copy-pasta a lot of setup / teardown for every test. I've simplified the tests so there are almost no nested context / describe blocks though to make it as human-readable as possible. Same for the comment below. The actual output is very clear:

```
     ✔  v2beta project rules API when the project does not exist GET /iam/v2beta/projects/:id/rules returns an empty list
```

Reply:
Oh right-- that context was not part of the commit so I glazed over it.

I understand the value of the context block and was not commenting on that previously. Feel free to just point out the line I didn't see 👍